diff --git a/node_helper.js b/node_helper.js
index 679dee9..04f1066 100644
--- a/node_helper.js
+++ b/node_helper.js
@@ -11,7 +11,7 @@ class Fetcher {
     this.timeer = null
     this.vevents = ''
 
-    var serveURL = `/CALDAV/${this.config?.name}.ics`
+    var serveURL = `/CALDAV/${this.config.name}.ics`
 
     expressApp.get(serveURL, (req, res) => {
       var filePath = path.resolve(__dirname, 'public', this.config.name + '.ics')
@@ -47,7 +47,7 @@ class Fetcher {
     const url = new URL(serverUrl)
     var urls = []
     const calendars = await client.fetchCalendars()
-    const calendarFilter = this.config?.calendarFilter || []
+    const calendarFilter = this.config.calendarFilter || []
     for (let calendar of calendars) {
       if (Array.isArray(calendarFilter) && calendarFilter.length > 0 && !calendarFilter.includes(calendar.displayName)) continue
       const objects = await client.calendarQuery({
