diff --git a/MMM-Saint.js b/MMM-Saint.js
index 8a9f830..3081bf6 100644
--- a/MMM-Saint.js
+++ b/MMM-Saint.js
@@ -58,7 +58,7 @@ Module.register("MMM-Saint", {
   displayData () {
     var today = document.getElementById("TODAY");
     var tomorrow = document.getElementById("TOMORROW");
-    if (this.config.personalize.displayIcon) today.innerHTML = `<span class="fas fa-bible"></span> ${this.Saint.today}`;
+    if (this.config.personalize.displayIcon) today.innerHTML = `<span class="fas fa-crown"></span> ${this.Saint.today}`;
     else {
       today.textContent = `${this.config.personalize.todayText} ${this.Saint.today}`;
       if (this.config.personalize.displayTomorrow) tomorrow.textContent = `${this.config.personalize.tomorrowText} ${this.Saint.tomorrow}`;
