diff --git a/MMM-VigilanceMeteoFrance.js b/MMM-VigilanceMeteoFrance.js
index 5539eba..7be3b80 100644
--- a/MMM-VigilanceMeteoFrance.js
+++ b/MMM-VigilanceMeteoFrance.js
@@ -276,6 +276,9 @@ Module.register("MMM-VigilanceMeteoFrance", {
 
 			this.sendSocketNotification('CONFIG', this.config);
 		}
+		if (notification === "MODULE_DOM_CREATED") {
+			this.sendSocketNotification('CONFIG', this.config);
+		}
 	},
 
 	// Use the received data to set the various values before update DOM
diff --git a/node_helper.js b/node_helper.js
index c234443..8651852 100644
--- a/node_helper.js
+++ b/node_helper.js
@@ -64,7 +64,6 @@ module.exports = NodeHelper.create({
 		var self = this;
 		if (notification === 'CONFIG') {
 			self.config = payload;
-			self.sendSocketNotification("STARTED", true);
 			self.getData();
 		}
 	}
